### DISCUSSION TITLE
lsp_workspace_symbols only show 100 results

### DISCUSSION THREAD

**Discussion by linepi** [ibhagwan/fzf-lua#1359] [Q&A]
![image](https://github.com/user-attachments/assets/cbd2a5e1-f903-4b02-bb98-cc9be4107b17)
Anyone knows why?

**Comment by ibhagwan** [comment]
Perhaps a configuration issue with your LSP server, mine shows way over 100 with `lua_ls`:
![IMG_2346](https://github.com/user-attachments/assets/970c7d1f-ed6d-4f1c-8c1d-ad539c3c2417)

**Reply by linepi** [reply]
Strangely enough, my lsp_document_symbol works fine. Can you give me some ideas on where to find the error?

**Reply by ibhagwan** [reply]
> Strangely enough, my lsp_document_symbol works fine. Can you give me some ideas on where to find the error?

Document symbol is a different query to the LSP, I’m not sure if it’s an error as the possibility it could be a LSP config option which would depend on which language server you’re using.

If you want to rule out fzf-lua you can try a different combo of language / LSP server.

**Comment by cpcopy** [comment]
On my setup, it also only returns 100 results with lsp_workspace_symbols, no matter FzfLua or Telescope. I think it is due to lsp(clangd) limitation. With lsp_live_workspace_symbols, it can search all symbols, though it only returns maximum 100 results.

One difference of lsp_workspace_symbols I found between FzfLua and Telescope is the behavior of lsp_workspace_symbols query=xxx. Most of the time, telescope can return the result of xxx, but FzfLua not (lsp_live_workspace_symbols works fine.). Is there any implementation difference?

**Reply by ibhagwan** [reply]
> One difference of lsp_workspace_symbols I found between FzfLua and Telescope is the behavior of lsp_workspace_symbols query=xxx. Most of the time, telescope can return the result of xxx, but FzfLua not (lsp_live_workspace_symbols works fine.). Is there any implementation difference?

`query` is reserved as fzf’s query (prompt input text) p, if you want to send a custom LSP request use `lsp_query`, you can even specify both (lsp query and fuzzy query).

**Reply by cpcopy** [reply]
Thanks. Got it.
